ruma-events/src/custom.rs

```diff
@@ -8,6 +8,40 @@ use crate::{
     RoomEventContent, StateEventContent,
 };
 
+/// A custom event that has been redacted.
+#[derive(Clone, Debug, Serialize)]
+pub struct RedactedCustomEventContent;
+
+impl EventContent for RedactedCustomEventContent {
+    fn event_type(&self) -> &str {
+        // TODO how do we handle this?
+        "custom"
+    }
```

Comment:
Should the `RedactedCustomEvent` be `RedactedCustomEvent { event_type: String }`?

Reply:
Yes. Also with `#[serde(skip)]` though.

Reply:
Without altering the `Event` derive this is what I came up with for the serialized redacted custom event

```rust
let redacted = json!({
    "content": {},
    "event_id": "$h29iv0s8:example.com",
    "origin_server_ts": 1,
    "sender": "@carl:example.com",
    "state_key": "hello there",
    "unsigned": unsigned,
    "type": "m.made.up"
});
```

The `Event` derive only skips fields if they are zst's, not sure how or if we should check if the event is custom?

Reply:
I guess we need another event content trait method. `fn has_fields() -> bool` should work.